src/components/AppContainer.tsx

```diff
@@ -595,6 +651,30 @@ export class AppContainer extends React.PureComponent<{}, IAppState> {
         return list[index];
     }
 
+    private selectGenderRestrictedWord<T extends {}>(list: T[], database : IWordDatabase, selectedGenders: Set<number>): T | undefined {
+        if (list.length === 0) {
+            return undefined;
+        }
+        // Could possibly filter based on the gender conditions, but this avoids having to do extra work.
```

Comment:
I think this amount of work is fine - I'd unite this with selectRandom to have just a single function, prefilter the list based on the genders, and then select randomly from the filtered list.